source/postprocess/geoid.cc

```diff
@@ -889,11 +980,11 @@ namespace aspect
     Geoid<dim>::required_other_postprocessors() const
     {
       std::list<std::string> deps;
-      if (include_dynamic_topo_contribution == true)
-        {
-          deps.emplace_back("dynamic topography");
-          deps.emplace_back("boundary densities");
-        }
+      if ( (include_surface_topo_contribution == true && use_real_surface_topography == false) || (include_CMB_topo_contribution == true && use_real_CMB_topography == false) )
```

Comment:
I'm not sure about this one, but could imagine that required_other_postprocessor is called before execute, in which case you don't know yet if use_real_surface_topography and use_real_CMB_topography are true or false. I will defer this one to @gassmoeller.

Reply:
You are right. I can't use use_real_surface_topography and use_real_CMB_topography. Not sure how to solve this differently

Reply:
My suggestion above will fix this. Both `parse_parameters` and `initialize` are called before `required_other_postprocessors`.